CMakeLists.txt

```
@@ -63,8 +64,29 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
+set(LATEST_Z3_VERSION "4.8.12")
+set(LOWEST_Z3_VERSION_SUPPORTED "4.8.0")
+find_package(Z3)
+if (${USE_Z3} AND ${Z3_FOUND})
```

`find_package()` already checks `USE_Z3` so you can reduce this to:
```suggestion
if (${Z3_FOUND})
```

---

done